src/java/org/apache/cassandra/service/StorageProxy.java

```diff
@@ -468,46 +468,54 @@ public static RowIterator cas(String keyspaceName,
 
             Supplier<Pair<PartitionUpdate, RowIterator>> updateProposer = () ->
             {
-                // read the current values and check they validate the conditions
-                Tracing.trace("Reading existing values for CAS precondition");
-                SinglePartitionReadCommand readCommand = (SinglePartitionReadCommand) request.readCommand(nowInSeconds);
-                ConsistencyLevel readConsistency = consistencyForPaxos == ConsistencyLevel.LOCAL_SERIAL ? ConsistencyLevel.LOCAL_QUORUM : ConsistencyLevel.QUORUM;
+                long startTimeNanos = System.nanoTime();
```

Comment:
I bet the `readOne` invocation below updates user read metrics. Should we avoid bumping read metrics since we now have a dedicated metric for this part of code? At very least we should clear in dashboard tooltips what's the relation between this metric and user read metric.

Reply:
I will update the tooltips and keep the read metrics behavior the same